lib/engine/tile.rb

```diff
@@ -328,6 +328,11 @@ def revenue_to_render
       @revenue_to_render ||= stops.map(&:revenue_to_render)
     end
 
+    # Used to set label for a recently placed tile
+    def label!(label_name)
```

Comment:
instead this should be a setter function

def label=(name)
  @label = ...
end

Reply:
Have now updated according to this. Also did some other corrections in the commit.